python/cudf/cudf/tests/test_binops.py

```diff
@@ -320,13 +320,35 @@ def test_series_compare_nulls(cmpop, dtypes):
     utils.assert_eq(expect, got)
 
 
+def string_series_compare_test_cases():
+    cases = []
+    pd_sr = pd.Series(["a", "b", None, "d", "e", None], dtype="string")
+    all_cmpop_cases = [
+        (pd_sr, pd_sr),
+        (pd_sr, "a"),
+        ("a", pd_sr),
+    ]
+
+    for op in _cmpops:
+        for case in all_cmpop_cases:
+            cases.append((*case, op))
+
+    eq_neq_cases = (
+        (pd_sr, 1),
+        (1, pd_sr),
+        (pd_sr, 1.5),
+        (1.5, pd_sr),
+        (pd_sr, True),
+        (True, pd_sr),
+    )
+    for case in eq_neq_cases:
+        cases += [(*case, operator.eq), (*case, operator.ne)]
+
+    return cases
```

Comment:
Could you help me understand a bit more why this approach is better? I've worked it up locally a few different ways and it seems like this forces some kind of programmatic skipping inside the test body since we're not dealing with a cartesian product of the parameterization, whereas the current approach just generates the exact tests we need. Is it the test output that you're concerned about here? (I think this would show up as `test_string_series_compare[data0]` on the command line for instance as opposed to an expanded output listing the actual parameters)

Reply:
We're trying to move away from a setup where test _collection_ creates heavyweight objects (e.g. the series here).

I see that you don't actually need the cartesian product. Does it work to separate the eq/neq cases from the other?

You can then have two test functions that handle the general case and then the eq/neq case taking different parameters. There's some discussion here https://docs.rapids.ai/api/cudf/nightly/developer_guide/testing.html